source/site-administrators/studio/rte2-setup.rst

```diff
@@ -148,6 +163,112 @@ You can also add custom elements to the rule set and can be done by simply addin
       :width: 85%
       :align: center
 
+   |
+
+.. _adding-external-plugins:
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Adding External Plugins
+^^^^^^^^^^^^^^^^^^^^^^^
+
+TinyMCE provides an option to specify URLS to plugins outside the tinymce plugins directory.  These external plugins allow the user to extend TinyMCE.  For example, you can create custom dialogs, buttons, menu items, etc.
+
+For more information on the Tiny MCE external_plugins option, see https://www.tiny.cloud/docs/configure/integration-and-setup/#external_plugins
+
+The Crafter Studio user does not have full control of the tinymce initialization.  To add a custom button to the toolbar in Crafter Studio, it would be done using the external plugin route since, what TinyMCE docs advise is not viable in Studio without doing a whole form control plugin where they'd have full control of tinymce initialization.
+
+To add an external plugin, use the tag ``<external_plugins />`` in the RTE configuration.
+Use the Crafter Studio API that gets a file for a given plugin, the getPluginFile API found here https://app.swaggerhub.com/apis/craftercms/studio/3.1.14.0#/plugin/getPluginFile to get the Tiny MCE external plugin file to pass to the RTE.
+
+Example External Plugin
+^^^^^^^^^^^^^^^^^^^^^^^
+Let's take a look at an example of a simple external plugin that creates a custom button which inserts text in the RTE.
+We'll load our external plugin (a custom button) and add it to the RTE's toolbar.  For our example, we'll be using a site created using the empty blueprint named ``hello``.
+
+1. Open the RTE (TinyMCE 5) configuration file in Studio by opening the **Sidebar**, then click on |siteConfig| -> *Configuration* -> *RTE (TinyMCE 5) Configuration*
+
+2. We'll add the configuration for TinyMCE to load the plugin using Crafter Studio's getPluginFile API. We achieve this by using the ``<external_plugins />`` tag and adding child tags with the id of the plugin as tag name and the target URL as the tag's content |br|
+
+   .. code-block:: xml
+
+      <external_plugins>
+        <my_button><![CDATA[/studio/api/2/plugin/file?siteId=hello&type=tinymce&name=my_button&filename=plugin.min.js]]></my_button>
+      </external_plugins>
+
+   |
+
+3. Add the custom button we're creating to the toolbar of the RTE.  Scroll to the ``<toolbarItems2 />`` tag and add the custom button we are creating ``my_button``
+
+   .. code-block:: xml
+
+      <toolbarItems2>my_button</toolbarItems2>
+
+   |
+
+4. Finally, we'll create our plugin file and add it in to Studio.  See :ref:`studio-plugins` for more information on creating a Crafter Studio plugin.
+
+   * Using information from step 2 for our external plugin, create the required directory structure for the plugin file, then create our plugin file named ``plugin.min.js``
+
+     .. code-block:: js
+        :linenos:
+        :caption: *$CRAFTER_HOME/data/repos/sites/SITE_NAME/sandbox/config/studio/plugins/tinymce/my_button/plugin.min.js*
```

Comment:
The .min.js typically refers to a minified file. This would be achieved by running the source file through something like `uglify`. 

Developers would create plugin.js, then run uglify to produce plugin.min.js. This is not strictly necessary but an optimization. For the docs, I recommend writing it all up to this point with just `plugin.js` and at the end of the working plugin docs, recommend people to minimize and if done, change the external_plugins > my_button url to load the minified version.